pkg/executables/kubectl.go

```diff
@@ -295,6 +295,11 @@ func (k *Kubectl) WaitForControlPlaneReady(ctx context.Context, cluster *types.C
 	return k.Wait(ctx, cluster.KubeconfigFile, timeout, "ControlPlaneReady", fmt.Sprintf("%s/%s", capiClustersResourceType, newClusterName), constants.EksaSystemNamespace)
 }
 
+// WaitForFirstControlPlaneReady blocks until the first control plane node is ready.
+func (k *Kubectl) WaitForFirstControlPlaneReady(ctx context.Context, cluster *types.Cluster, timeout string, newClusterName string) error {
+	return k.Wait(ctx, cluster.KubeconfigFile, timeout, "ControlPlaneInitialized", fmt.Sprintf("%s/%s", capiClustersResourceType, newClusterName), constants.EksaSystemNamespace)
```

Comment:
What's the rationale between using `ControlPlaneInitialized` instead of `ControlPlaneReady`?

`ControlPlaneReady` comes after `ControlPlaneInitialized` and ensures the CP is "contactable", which I think suits better our usecase. `ControlPlaneInitialized` only ensures that the kubeconfig exists.

If we have a good reason to change it, I would suggest documenting it at least with a comment, since it's a change in behavior in one of core pieces of logic.

Reply:
That's not quite accurate.

> // ControlPlaneInitializedCondition reports if the cluster's control plane has been initialized such that the
> // cluster's apiserver is reachable and at least one control plane Machine has a node reference.

The change is necessary because we need to install the CNI as quickly as possible to facilitate our `MachineHealthChecks` that expect the `Ready` condition (nodes won't be ready until they have a CNI installed).